packages/contracts/src/executor/QueryExecutor.ts

```diff
@@ -0,0 +1,50 @@
+import { GenericContractQueryCall, ContractMessage, ContractOptions } from '@dedot/contracts';
+import { GenericSubstrateApi } from '@dedot/types';
+import { assert, concatU8a, hexToU8a, stringCamelCase, u8aToHex } from '@dedot/utils';
+import { Executor } from './Executor.js';
+
+export class QueryExecutor<ChainApi extends GenericSubstrateApi> extends Executor<ChainApi> {
+  doExecute(message: string): GenericContractQueryCall {
+    const messageMeta = this.#findMessage(message);
+
+    assert(messageMeta, `Query message not found: ${message}`);
+
+    const callFn: GenericContractQueryCall = async (...params: any) => {
+      const { args } = messageMeta;
+
+      let caller = params[args.length] || this.address;
```

Comment:
I'm not sure we want to have this default behavior, we probably should user decide the `caller` address